suppkg_rebuild.py

```diff
@@ -0,0 +1,255 @@
+
+    def get_support_package_list(self, project, repository):
+        f = osc.core.get_buildconfig(self.apiurl, project, repository).splitlines()
+        pkg_list = []
+        for line in f:
+            if re.match('Preinstall', line) or re.match('VMinstall', line):
+                content = line.split(':')
+                variables = [x.strip() for x in content[1].split(' ')]
+                for var in variables:
+                    if var != '' and var not in pkg_list:
+                        if var.startswith('!') and var[1:] in pkg_list:
+                            pkg_list.remove(var[1:])
+                        else:
+                            pkg_list.append(var)
+        return pkg_list
+
+    def get_project_binarylist(self, project, repository, arch):
+        query = {'view': 'binarylist', 'repository': repository, 'arch': arch}
+        root = ET.parse(http_GET(makeurl(self.apiurl,['build', project, '_result'],
+                                 query=query))).getroot()
+        return root
+
+    def is_repo_dirty(self, project, repository):
+        url = makeurl(self.apiurl, ['build', project, '_result?view=summary&repository=%s' % repository])
+        root = ET.parse(http_GET(url)).getroot()
+        for repo in root.findall('result'):
+            repostate = repo.get('state', 'missing')
+            if repostate not in ['unpublished', 'published'] or repo.get('dirty', 'false') == 'true':
+                return True 
+        return False
+
+    def get_package_buildinfo(self, project, repository, arch, package):
+        url = makeurl(self.apiurl,['build', project, repository, arch, package, '_buildinfo'])
+        root = ET.parse(http_GET(url)).getroot()
+
+        return root
+
+    def get_buildinfo_version(self, project, package):
+        buildinfo = self.get_package_buildinfo(project, 'standard', 'x86_64', package)
+        versrel = buildinfo.find('versrel')
+        version = versrel.split('-')[0]
+
+        return version
+
+    def process_project_binarylist(self, project, repository, arch):
+        prj_binarylist = self.get_project_binarylist(project, repository, arch)
+        files = {}
+        for package in prj_binarylist.findall('./result/binarylist'):
+            for binary in package.findall('binary'):
+                result = re.match(r'(.*)-([^-]*)-([^-]*)\.([^-\.]+)\.rpm', binary.attrib['filename'])
+                if not result:
+                    continue
+                bname = result.group(1)
+                if bname.endswith('-debuginfo') or bname.endswith('-debuginfo-32bit'):
+                    continue
+                if bname.endswith('-debugsource'):
+                    continue
+                if bname.startswith('::import::'):
+                    continue
+                if result.group(4) == 'src':
+                    continue
+                files[bname] = package.attrib['package']
+
+        return files
+
+    def get_staging_projects(self, project):
```

Comment:
More code that's in `stagingapi` minus the filter to letter, but that code is also in `get_staging_projects_short()`.